addons/mail_group/models/mail_group.py

```diff
@@ -251,6 +251,11 @@ def _alias_get_error_message(self, message, message_dict, alias):
         self.ensure_one()
 
         if alias.alias_contact == 'followers':
+            # Selected group of users
+            email = email_normalize(message_dict.get('email_from'))
+            if self.access_mode == 'groups' and \
+               self.search([('id', '=', self.id), ('access_group_id.users.email_normalized', '=', email)]):
```

Comment:
- lightening done :heavy_check_mark: 
- came up with a better error message (assumed you referred to the test error msg)

> And shouldn't we avoid checking members if that's not the right mode? As currently it will always check whether you're a member if you're not part of the group in "group mode"

Not sure I understood that. I think you refer to this part. Do you think of something like that:
```diff
 if alias.alias_contact == 'followers':  #  alias.alias_contact set to "everyone" when mail is allowed from 
            # Selected group of users
            email = email_normalize(message_dict.get('email_from'))
            if self.access_mode == 'groups' and \
               self.search_count([('id', '=', self.id), ('access_group_id.users.email_normalized', '=', email)], limit=1):
                return  # Skip the verification because the partner is in the user group list
            # Members only
-            if not self._find_member(message_dict.get('email_from')):
+            if self.access_mode == 'members' and not self._find_member(message_dict.get('email_from')):
                return _('Only members can send email to the mailing list.')
            # Skip the verification because the partner is in the member list
            return
```

In a situation where everyone can send mails to mail group, we won't go into this line:
`if self.access_mode == 'members' and not self._find_member(message_dict.get('email_from')):` 
because alias.alias_contact would be set to "everyone". Because of that we don't need to apply the above diff.

@reth-odoo let me know if I got you correctly and if that makes sense ;)